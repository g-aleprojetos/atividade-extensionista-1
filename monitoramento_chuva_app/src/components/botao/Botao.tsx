import React from 'react';
import * as S from './Botao.styles';

export type Props = S.PropsBotao & {
  children?: React.ReactNode;
  handleOnClick?: () => void;
};

export const Botao = (props: Props) => {
  const {tipo, children, handleOnClick} = props;

  return (
    <S.Container data-testid={'test_botao_container'}>
      <S.ContainerBotao
        tipo={tipo}
        type={tipo === 'submit' ? 'submit' : 'button'}
        onClick={handleOnClick}>
        {children}
      </S.ContainerBotao>
    </S.Container>
  );
};
