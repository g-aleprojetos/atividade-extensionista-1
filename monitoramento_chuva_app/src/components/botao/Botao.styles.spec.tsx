import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect'
import 'jest-styled-components'
import * as S from './Botao.styles';

describe('Botao.styles', () => {
  test('Container DEVE ser igual ao snapshot', () => {
    const tree = renderer.create(<S.Container/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('ContainerBotao DEVE ser igual ao snapshot', () => {
    const tree = renderer.create(<S.ContainerBotao/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
