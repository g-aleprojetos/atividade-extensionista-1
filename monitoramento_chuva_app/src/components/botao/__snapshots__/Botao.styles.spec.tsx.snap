// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Botao.styles Container DEVE ser igual ao snapshot 1`] = `
.c0 {
  margin: 15px;
}

<div
  className="c0"
/>
`;

exports[`Botao.styles ContainerBotao DEVE ser igual ao snapshot 1`] = `
.c0 {
  background-color: #04c582;
  border-radius: 8px;
  border-style: none;
  padding: 8px 60px;
  cursor: pointer;
}

<button
  className="c0"
/>
`;
